### PR TITLE
Add support for foreign keys to the Forge

### PR DESCRIPTION
**Pull request by davidgv88**
Add support for foreign keys to the Forge

### CODE REVIEW

**Comment by davidgv88** [comment]
@lonnieezell If you like I can write the documentation and Unit tests.

You can see an example in the Checks controller and forge method.

**Comment by jim-parry** [comment]
not to be too picky, but we expect commits to be GPG-signed :)

File: system/Database/Forge.php [hunk 1]
```diff
@@ -429,8 +491,12 @@ protected function _createTable($table, $if_not_exists, $attributes)
 				: "\n\t".$this->_processColumn($columns[$i]);
 		}
 
-		$columns = implode(',', $columns)
-			.$this->_processPrimaryKeys($table);
+                
+		$columns = implode(',', $columns);
+                
+                $columns .= $this->_processPrimaryKeys($table);
```
Comment (davidgv88) [inline system/Database/Forge.php]: Rename _processPrimaryKeys to what name?

File: system/Database/Forge.php [hunk 2]
```diff
@@ -330,6 +337,61 @@ public function addField($field)
 	}
 
 	//--------------------------------------------------------------------
+        
+        /**
+	 * Add Foreign Key
+	 *
+	 * @param    array $field
+	 *
+	 * @return    CI_DB_forge
```
Comment (davidgv88) [inline system/Database/Forge.php]: In other methods (addField, addKeys) is the same.

Whats is the correct return?  \CodeIgniter\Database\Forge

This return \CodeIgniter\Database\Postgre\Forge or \CodeIgniter\Database\MySQLi\Forge

File: system/Database/Forge.php (hunk 1 above)
Comment (lonnieezell) [inline system/Database/Forge.php]: Just removing the _ should be good: `processPrimaryKeys`. I didn't see a method name that would conflict but just did a quick search through the changes.

**Comment by lonnieezell** [comment]
OH - and FYI - the Checks  controller isn't official testing, just something I have tendency to use while working on a feature that gives me something I can see outside of testing. I accidentally committed it once and have kept it in for now because I figure it helps give alpha testers an idea how to use certain features. 

But the testing we need would be more PHPUnit tests, specifically live database tests. See the existing ones at `tests/system/Database/Live`.

**Comment by davidgv88** [comment]
@lonnieezell I know sorry! The next week I do the PHP Unit Tests.

The forge method in Checks controller will be removed!

**Comment by lonnieezell** [comment]
Oh, that's fine. I don't mind them being there, not at all. Just making sure you knew, which I figured you did. :)

**Comment by davidgv88** [comment]
Sorry @lonnieezell . Can you delete this PR?

I will create another PR

I have some problems with merge xD.

Sorry I do not have much experience in collaborating

**Pull request closed without merging**